lib/MetamodelX/Red/Relationship.pm6

```diff
@@ -75,7 +75,7 @@ multi method add-relationship(::Type Mu:U $self, Red::Attr::Relationship $attr)
             $attr.package.^rs.new: :filter($ast)
         }
         !! my method (Mu:U:) {
-            if ($*RED-GREP-FILTER ~~ Red::AST).Bool {
+            if ($*RED-GREP-FILTER && $*RED-GREP-FILTER ~~ Red::AST).Bool {
```

Comment:
that's not true nor defined here...

Reply:
That's the thing with the dynamic variables, if they haven't been defined elsewhere then they have a value which is a Failure, so whilst the original match wouldn't return true in this case (a Failure !~~ Red::AST,) you will get the warning about the Failure.  But you can safely use the Failure in a boolean context like that to stop the warning.

Reply:
I've fixed it with this: e60fc32f7632e49b93bd64472faf7a1f3ec9ac1f would you mind to review?

Reply:
Yep that looks fine :-)